spec/services/exports/export_distributions_csv_service_spec.rb

```diff
@@ -75,88 +167,63 @@
-            {quantity: 0, value: Money.new(0), item_id: nil},
-            {quantity: 8, value: Money.new(2400), item_id: 1}
-          ],
-          [
-            {quantity: 0, value: Money.new(0), item_id: nil},
-            {quantity: 1, value: Money.new(1000), item_id: 3},
-            {quantity: 0, value: Money.new(0), item_id: nil},
-            {quantity: 0, value: Money.new(0), item_id: nil},
-            {quantity: 0, value: Money.new(0), item_id: nil}
-          ],
-          [
-            {quantity: 0, value: Money.new(0), item_id: nil},
-            {quantity: 0, value: Money.new(0), item_id: nil},
-            {quantity: 2, value: Money.new(4000), item_id: 4},
-            {quantity: 0, value: Money.new(0), item_id: nil},
-            {quantity: 0, value: Money.new(0), item_id: nil}
-          ],
-          [
-            {quantity: 0, value: Money.new(0), item_id: nil},
-            {quantity: 0, value: Money.new(0), item_id: nil},
-            {quantity: 0, value: Money.new(0), item_id: nil},
-            {quantity: 3, value: Money.new(9000), item_id: 5},
-            {quantity: 0, value: Money.new(0), item_id: nil}
-          ]
+          "A Item", "A Item In-Kind Value",
+          "B Item", "B Item In-Kind Value",
+          "C Item", "C Item In-Kind Value",
+          "Dupe Item", "Dupe Item In-Kind Value",
+          "E Item", "E Item In-Kind Value"
         ]
-      end
+      }
+      let(:expected_headers) { non_item_headers + item_headers }
 
       it 'should match the expected content with in-kind value of each item for the csv' do
         allow(organization).to receive(:include_in_kind_values_in_exported_files).and_return(true)
         expect(subject[0]).to eq(expected_headers)
 
-        distributions.zip(expected_items).each_with_index do |(distribution, expected_item), idx|
+        expected_distributions.zip(expected_items).each_with_index do |(expected_dist, expected_item), idx|
           row = [
-            distribution.partner.name,
-            distribution.created_at.strftime("%m/%d/%Y"),
-            distribution.issued_at.strftime("%m/%d/%Y"),
-            distribution.storage_location.name,
-            distribution.line_items.where(item_id: item_id).total,
-            Money.new(distribution.line_items.where(item_id: item_id).total_value).to_f,
-            distribution.delivery_method,
-            "$#{distribution.shipping_cost.to_f}",
-            distribution.state,
-            distribution.agency_rep,
-            distribution.comment,
+            partner.name,
+            expected_dist[:created_at],
+            expected_dist[:issued_at],
+            storage_location.name,
+            expected_dist[:filtered_quantity],
+            expected_dist[:filtered_total_value],
+            expected_dist[:delivery_method],
+            "$#{expected_dist[:shipping_cost].to_f}",
+            expected_dist[:state],
+            expected_dist[:agency_rep],
+            expected_dist[:comment],
```

Comment:
Is there a reason you can't just generate a flat CSV string and compare the results against it instead of going row by row?

Reply:
This comment still stands - see e.g. https://github.com/rubyforgood/human-essentials/blob/d8fa236212a4114d962d4f7ac1002f42ba058c2e/spec/requests/items_requests_spec.rb#L52-L52